dataflux_pytorch/benchmark/checkpointing/multinode/train_async_save.py

```diff
@@ -0,0 +1,141 @@
+import os
+import statistics
+import time
+
+import torch
+import torch.distributed as dist
+from lightning import Trainer
+from lightning.pytorch.callbacks import ModelCheckpoint
+from lightning.pytorch.demos import (LightningTransformer, Transformer,
+                                     WikiText2)
+from torch.utils.data import DataLoader
+
+from demo.lightning.checkpoint.multinode.strategies import DatafluxFSDPStrategy
+from demo.lightning.checkpoint.multinode.train import init_processes
+
+
+class DemoTransformer(LightningTransformer):
+
+    def __init__(
+        self,
+        vocab_size: int = 33278,
+        nlayers: int = 2,
+        optimizer: str = "sgd",
+        simulated_workload_duration: int = 0,
+    ) -> None:
+        super().__init__(vocab_size)
+        self.optimizer = optimizer
+        self.simulated_workload_duration = simulated_workload_duration
+        self.model = Transformer(vocab_size=vocab_size, nlayers=nlayers)
+
+    def configure_optimizers(self) -> torch.optim.Optimizer:
+        # Use self.trainer.model.parameters so that we can set
+        # use_orig_params=False on the Strategy. Using AdamW also results in a
+        # checkpoint size roughly 20% of used GPU memory.
```

Comment:
as in 20% of the GPU memory used for the training, or 20% of the GPU memory that is available for use?

Reply:
My understanding is that AdamW uses fewer parameters than SGD making it use less GPU memory for training. However, this optimizer choice is irrelevant to the distributed async save demo, so I've removed it.